addons/portal/models/portal_mixin.py

```diff
@@ -132,3 +132,17 @@ def get_portal_url(self, suffix=None, report_type=None, download=None, query_str
             '#%s' % anchor if anchor else ''
         )
         return url
+
+    def _get_portal_data_from_context(self):
+        """Returns the current portal partner and its associated thread records from the context,
+        if applicable."""
+        portal_data = self.env.context.get("portal_data", {})
+        portal_partner = portal_data.get("portal_partner")
+        portal_thread = portal_data.get("portal_thread")
+        if isinstance(portal_partner, self.pool["res.partner"]) and isinstance(
+            portal_thread, self.pool["mail.thread"]
+        ):
```

Comment:
This check is useless: it is the role of the caller to send the right data. This is mainly adding noise and making code hard to read.

Reply:
If someone puts something else, it is going to crash or make invalid comparisns. So what's the point ?